### PR TITLE
Added a function returning the clustering spectrum of a network

### PR DESCRIPTION
**Pull request by antoineallard**

### CODE REVIEW

File: dynamicalab/algorithms/clustering_spectrum.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      **Example**
+      
+
+      .. code:: python
+
+        import networkx as nx
+        import dynamicalab.algorithms as algo
+
+        G = nx.florentine_families_graph()
+        onion, kcore = algo.clustering_spectrum(G)
```
Comment (laurencee9) [inline dynamicalab/algorithms/clustering_spectrum.py]: I don't think this is correct.

File: dynamicalab/algorithms/clustering_spectrum.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      
+      .. warning::
+
+        The graph must be **simple** (no multiedges), **undirected** and **without self-loops**.
+      
+
+      **Retuns**
+
+      clust_spect : dict
+          Dictionary mapping each degree class to the corresponding average local clustering coefficient.
```
Comment (laurencee9) [inline dynamicalab/algorithms/clustering_spectrum.py]: Could be interesting to give an example of the dictionary (is the degree class an integer?)

File: dynamicalab/algorithms/clustering_spectrum.py (hunk 1 above)
Comment (antoineallard) [reply dynamicalab/algorithms/clustering_spectrum.py]: hahaha indeed

**Comment by antoineallard** [comment]
This should be fine now. I already changed docs/source/reference/algorithms/index.rst to include the new function.

**Pull request merged**